presto-main/src/main/java/com/facebook/presto/spiller/MultiPathBinaryFileSingleStreamSpillerFactory.java

```diff
@@ -68,6 +84,19 @@ private static ListeningExecutorService createExecutorServiceOfSize(int nThreads
     @Override
     public SingleStreamSpiller create(List<Type> types, LocalSpillContext localSpillContext)
     {
-        return new BinaryFileSingleStreamSpiller(blockEncodingSerde, executor, spillPath, spillerStats, localSpillContext);
+        return new BinaryFileSingleStreamSpiller(blockEncodingSerde, executor, getNextSpillPath(), spillerStats, localSpillContext);
+    }
+
+    private Path getNextSpillPath()
+    {
+        int spillPathsCount = spillPaths.size();
+        int startIndex = roundRobinIndex.incrementAndGet();
```

Comment:
I changed the `getNextSpillPath` to be synchronized and single call to it changes the `roundRobinIndex` to point to first spillPath after one which was picked.